flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/Fabric8FlinkKubeClient.java

```diff
@@ -191,7 +192,12 @@ public Optional<Endpoint> getRestEndpoint(String clusterId) {
 
     @Override
     public List<KubernetesPod> getPodsWithLabels(Map<String, String> labels) {
-        final List<Pod> podList = this.internalClient.pods().withLabels(labels).list().getItems();
+        final List<Pod> podList =
+                this.internalClient
+                        .pods()
+                        .withLabels(labels)
+                        .list(new ListOptionsBuilder().withResourceVersion("0").build())
```

Comment:
```suggestion
                        .list(new ListOptionsBuilder().withResourceVersion("0").build())
```
We'd better add some comments for this magic number.